R/tar-terra-rast.R

```diff
@@ -117,9 +117,9 @@ create_format_terra_raster <- function(filetype, gdal, ...) {
         terra::writeRaster(
             object,
             path,
-            filetype = NULL,
+            filetype = geotargets::geotargets_option_get("raster.gdal_driver_name"),
             overwrite = TRUE,
-            gdal = NULL
+            gdal = geotargets::geotargets_option_get("raster.gdal_creation_options")
```

Comment:
The user specified `filetype` and `gdal` option specified by the user are injected after the function is defined, on L125 and L126, so that will overwrite any options if the user has set something for a specific call. With the caveat that that seems to not work in {covr} context--I will investigate this further as it may be a sign of a bigger problem with this approach.

https://github.com/njtierney/geotargets/blob/5b71a2afaa096dbcbb06481d6ab498734b9bba50/R/tar-terra-rast.R#L125-L126

The order of operations is a little confusing, so would be happy to revise if there is a clearer way. 
Currently in `tar_terra_rast()` if NULL user input we pull the option and check it against available drivers with `match.arg()`. If user input is specified, that is also checked against the drivers. The function template is defined to use defaults for drivers, which includes env vars and options, ultimately falling back when those are not defined to is loaded into `geotargets.env` on attaching the package

https://github.com/njtierney/geotargets/blob/5b71a2afaa096dbcbb06481d6ab498734b9bba50/R/AAAA.R#L7-L10

So, the precedence for source of these custom options goes:

 - User input to target function > System environment variable > global option > package environment

The failures in {covr} appear to be related to the `NULL` template values not being replaced by `body<-` when they should be. This seems like it could be some sort of issue with evaluation environments, or something funky {covr} does that delays evaluation of the `body<-` call.

> Also we don't need to use geotargets::

You are right. I intended to protect against the package namespace not being loaded in the context that the `tar_format(write=)` function gets called, but I confirm it at least doesn't break our current tests to remove this inside the `.write_terra_raster()` function

Reply:
Note that it seems that {covr} does not like us not having `geotargets::` inside the write function (https://github.com/njtierney/geotargets/pull/19/commits/f634c301432549d20f9da928ba9771839828fe90). But you were right the other cases I was using it were unneeded!

Reply:
Awesome thanks for explaining! 

> User input to target function > System environment variable > global option > package environment

Makes sense to me, this sort of stuff is a bit curly, so I appreciate you unpacking it, and I think it's yet another reason to have this package to help remove some of this heavy lifting.